Algorithm/QCAlgorithm.Python.cs

```diff
@@ -841,7 +841,7 @@ public PyObject History(PyObject type, Symbol symbol, int periods, Resolution? r
             if (resolution == Resolution.Tick) throw new ArgumentException("History functions that accept a 'periods' parameter can not be used with Resolution.Tick");
 
             var res = GetResolution(symbol, resolution);
-            var start = _historyRequestFactory.GetStartTimeAlgoTz(symbol, periods, res, GetExchangeHours(symbol));
+            var start = _historyRequestFactory.GetStartTimeAlgoTz(symbol, periods, res, GetExchangeHours(symbol), GetDataTimeZone(symbol));
```

Comment:
I think in these other couple of cases, where there is no config, we could replace `GetExchangeHours` & `GetDataTimeZone` for a single get of the MHDB.Entry which has both data we need